Celeste.Mod.mm/Patches/AudioState.cs

```diff
@@ -1,9 +1,50 @@
-﻿namespace Celeste {
+﻿#pragma warning disable CS0626 // Method, operator, or accessor is marked external and has no attributes on it
+
+using MonoMod;
+
+namespace Celeste {
     class patch_AudioState : AudioState {
 
+        public float AmbienceVolume;
+
+        [MonoModConstructor]
+        public patch_AudioState() {
+            AmbienceVolume = 1f;
+        }
+
+        [MonoModConstructor]
+        public patch_AudioState(AudioTrackState music, AudioTrackState ambience) {
+            if (music != null) {
+                Music = music.Clone();
+            }
+            if (ambience != null) {
+                Ambience = ambience.Clone();
+            }
+            AmbienceVolume = 1f;
+        }
+
+        [MonoModConstructor]
+        public patch_AudioState(string music, string ambience) {
+            Music.Event = music;
+            Ambience.Event = ambience;
+            AmbienceVolume = 1f;
+        }
```

Comment:
initially i had `[MonoModReplace]` on these as i'm not calling the orig constructors, but that caused MiniInstaller to crash with a cryptic error.
i'd prefer to use `[MonoModReplace]` here, so unless that's impossible i'd prefer to hold off on merging this